aten/src/ATen/native/LossCTC.cpp

```diff
@@ -364,6 +365,7 @@ Tensor ctc_loss(const Tensor& log_probs, const Tensor& targets, IntArrayRef inpu
     res = std::get<0>(at::_ctc_loss(log_probs, targets, input_lengths, target_lengths, BLANK, zero_infinity));
     if (zero_infinity) {
       res = at::where(res == Scalar(std::numeric_limits<double>::infinity()), at::zeros({}, res.options()), res);
+      res = at::where(res != res, at::zeros({}, res.options()), res);
```

Comment:
Similar to the above, this is on the output of the forward, so I'm not sure it matches your observation.

Reply:
similar comments as above.

Reply:
But this is about forward vs. backward, not about CPU vs. GPU.

Reply:
Per your comment below, I think this would not have any effect (until the inputs are NaN), so I think we don't need it.